routes/dao_coin_exchange.go

```diff
@@ -0,0 +1,183 @@
+	}
+
+	if requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername != "" {
+		coin2ProfilePublicKeyBytes, _, err = fes.GetPubKeAndProfileEntryForUsernameOrPublicKeyBase58Check(
+			requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername,
+			utxoView,
+		)
+		if err != nil {
+			_AddBadRequestError(
+				ww,
+				fmt.Sprint("GetDAOCoinLimitOrders: Invalid DAOCoin2CreatorPublicKeyBase58CheckOrUsername"),
+			)
+			return
+		}
+		coin2PKID = utxoView.GetPKIDForPublicKey(coin2ProfilePublicKeyBytes).PKID
+		coin2ProfilePublicBase58Check = lib.Base58CheckEncode(coin2ProfilePublicKeyBytes, false, fes.Params)
+	}
+
+	adapter := utxoView.GetDbAdapter()
+
+	ordersBuyingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin1PKID, coin2PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	ordersSellingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin2PKID, coin1PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	var response []DAOCoinLimitOrder
+
+	for _, order := range ordersBuyingCoin1 {
+		transactorPublicKey := utxoView.GetPublicKeyForPKID(order.TransactorPKID)
+		response = append(response, DAOCoinLimitOrder{
+			TransactorPublicKeyBase58CheckOrUsername: lib.Base58CheckEncode(transactorPublicKey, false, fes.Params),
+
+			BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername:  coin1ProfilePublicBase58Check,
+			SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername: coin2ProfilePublicBase58Check,
+			ScaledExchangeRateCoinsToSellPerCoinToBuy:           order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			ExchangeRateCoinsToSellPerCoinToBuy: floatExchangeRateCoinsToSellPerCoinToBuy(
+				order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			),
+			QuantityToBuyInBaseUnits: order.QuantityToBuyInBaseUnits,
+			QuantityToBuy:            floatQuantityToBuy(order.QuantityToBuyInBaseUnits),
+		})
+	}
+
+	for _, order := range ordersSellingCoin1 {
+		transactorPublicKey := utxoView.GetPublicKeyForPKID(order.TransactorPKID)
+		response = append(response, DAOCoinLimitOrder{
+			TransactorPublicKeyBase58CheckOrUsername: lib.Base58CheckEncode(transactorPublicKey, false, fes.Params),
+
+			BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername:  coin2ProfilePublicBase58Check,
+			SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername: coin1ProfilePublicBase58Check,
+			ScaledExchangeRateCoinsToSellPerCoinToBuy:           order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			ExchangeRateCoinsToSellPerCoinToBuy: floatExchangeRateCoinsToSellPerCoinToBuy(
+				order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			),
+			QuantityToBuyInBaseUnits: order.QuantityToBuyInBaseUnits,
+			QuantityToBuy:            floatQuantityToBuy(order.QuantityToBuyInBaseUnits),
+		})
+	}
+
+	_ = json.NewEncoder(ww).Encode(GetDAOCoinLimitOrdersResponse{Orders: response})
```

Comment:
we need to catch this error. 
```suggestion
	if err = json.NewEncoder(ww).Encode(res); err != nil {
		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Problem encoding response as JSON: %v", err))
		return
	}
```